src/main/healthCheck.ts

```diff
@@ -0,0 +1,50 @@
+import log from 'electron-log/main'
+import { HttpsProxyAgent } from 'https-proxy-agent'
+import https from 'node:https'
+
+import { getProxyCertificateContent, getProxyURL } from './proxy'
+
+export const checkProxyHealth = async () => {
+  try {
+    const results = await Promise.allSettled([
+      isUrlReachable('https://www.google.com/generate_204'),
+      isUrlReachable('https://quickpizza.grafana.com'),
+      isUrlReachable('https://k6.io'),
+    ])
```

Comment:
@e-fisher I'm now using three distinct domains for the health check.

`https://www.google.com/generate_204` is pinged by android devices to check for connectivity and I ended up leaving `quickpizza.grafana.com` as it's lightweight as well.

Any ideas of domains we could include/remove?

Reply:
I would probably remove k6.io or replace it with something else, considering its future is uncertain